### PR TITLE
Add some methods to AnnotationMirrorMap and AnnotationMirrorSet.

### PR DESCRIPTION
**Pull request by smillst**
Used by PR #2143.

### CODE REVIEW

File: framework/src/main/java/org/checkerframework/framework/util/AnnotationMirrorMap.java [hunk 1]
```diff
@@ -116,4 +125,9 @@ public void clear() {
     public Set<Entry<AnnotationMirror, V>> entrySet() {
         return shadowMap.entrySet();
     }
+
+    /** @return an empty annotation mirror map */
+    public static <V> AnnotationMirrorMap<V> emptyMap() {
```
Comment (smillst) [inline framework/src/main/java/org/checkerframework/framework/util/AnnotationMirrorMap.java]: Why is this needed?

File: framework/src/main/java/org/checkerframework/framework/util/AnnotationMirrorSet.java [hunk 2]
```diff
@@ -133,4 +133,16 @@ public boolean removeAll(Collection<?> c) {
...
         shadowSet.clear();
     }
+
+    /**
+     * Returns a new {@link AnnotationMirrorSet} that contains {@code value}.
+     *
+     * @param value AnnotationMirror to put in the set
+     * @return a new {@link AnnotationMirrorSet} that contains {@code value}.
+     */
+    public static AnnotationMirrorSet singleton(AnnotationMirror value) {
```
Comment (smillst) [inline framework/src/main/java/org/checkerframework/framework/util/AnnotationMirrorSet.java]: Wrong name.  Look at Set implementation.

Comment (smillst) [reply framework/src/main/java/org/checkerframework/framework/util/AnnotationMirrorSet.java]: This method has the same name as [Collections#singleton](
https://docs.oracle.com/javase/8/docs/api/java/util/Collections.html#singleton(T)).  So, I think it is the right name.  We could also call it `signletonSet`.

Comment (wmdietl) [reply framework/src/main/java/org/checkerframework/framework/util/AnnotationMirrorSet.java]: That method states: `Returns an immutable set containing only the specified object.`
So should this method also return something immutable?
Or the javadoc should highlight this difference to the standard API behavior.

Comment (smillst) [reply framework/src/main/java/org/checkerframework/framework/util/AnnotationMirrorSet.java]: Good point.  I changed it to `singleElement`.

**Comment by wmdietl** [comment]
Thanks, these extensions look good now.
